File: Task 1/C++/Components/Product/Accessory.cpp
#include "Accessory.h"

Accessory::Accessory() : Apparel(){
    brand = "";
}

Accessory::Accessory(std::string name, float cost, std::string supplier, float discount_amount, int packing_units, float cost_per_unit, std::string material, std::string brand) 
            : Apparel(name, cost, supplier, discount_amount, packing_units, cost_per_unit, material){
    this->brand = brand;
}

Accessory::Accessory(const Accessory& other) : Apparel(other) {
    this->brand = other.brand;
}

std::string Accessory::display_product_info(){
    return Apparel::display_product_info() 
            +"\nBrand: " +brand;
}

File: Task 1/Java/logistics/src/main/java/com/logistics/Components/Product/Accessory.java
package com.logistics.Components.Product;

public class Accessory extends Apparel {

    private String brand;

    public Accessory() {
        super();
        brand = "";
    }

    public Accessory(String name, float cost, String supplier, float discountAmount, int packingUnits, float costPerUnit, String material, String brand) {
        super(name, cost, supplier, discountAmount, packingUnits, costPerUnit, material);
        this.brand = brand;
    }

    public Accessory(Accessory other) {
        super(other);
        this.brand = other.brand;
    }

    @Override
    public Accessory clone() {
        return new Accessory(this);
    }

    // Getters
    public String getBrand() { return brand; }

    // Setters
    public void setBrand(String brand) { this.brand = brand; }

    @Override
    public String display_product_info() {
        return super.display_product_info() + 
                "\nBrand: " + brand;
    }
}
